src/common.f90

```diff
@@ -4,4 +4,79 @@ module m_common
    integer, parameter :: dp=kind(0.0d0)
    real(dp), parameter :: pi = 4*atan(1.0_dp)
 
+   type :: globs_t
+      integer :: nx, ny, nz
+      integer :: nx_loc, ny_loc, nz_loc
+      integer :: n_groups_x, n_groups_y, n_groups_z
+      real(dp) :: Lx, Ly, Lz
+      real(dp) :: dx, dy, dz
+      integer :: nproc_x = 1, nproc_y = 1, nproc_z = 1
+      character(len=20) :: BC_x_s, BC_x_e, BC_y_s, BC_y_e, BC_z_s, BC_z_e
+   end type globs_t
+
+contains
+
+   subroutine set_pprev_pnext(xprev, xnext, yprev, ynext, zprev, znext, &
+                              xnproc, ynproc, znproc, nrank)
+      implicit none
+
+      integer, intent(out) :: xprev, xnext, yprev, ynext, zprev, znext
+      integer, intent(in) :: xnproc, ynproc, znproc, nrank
+
+      integer :: i, ix, iy, iz
+
+      ! Set the pprev and pnext for each rank
+      i = 0
+      do iz = 1, znproc
+         do iy = 1, ynproc
+            do ix = 1, xnproc
+               if (nrank == i) then
```

Comment:
The 3 loops here could be avoided with something like:
```
ix = modulo(nrank, xnproc)
iy = modulo(nrank - ix*xnproc, ynproc)
iz = modulo(nrank - ix*xnproc - iy*ynproc*xnproc)
```

Reply:
Sorry, just realised I made a mistake as modulo goes from `0` to `xnprox-1`, so there is likely some `+1` to put in there somewhere. Emphasising the need for tests :)